tests/test_NoiseReplacer.py

```diff
@@ -104,6 +104,29 @@ def testNoiseReplacer(self, noiseSource, variance):
             # fail (indeed, 67% should)
             self.assertLess(record.get("test_NoiseReplacer_outside"), np.sqrt(sumVariance))
 
+    @methodParameters(noiseSource=['measure', 'variance', 'meta', 'variance_median'],
+                      variance=[1.0, 1.0, 2.0, 1.0])
+    def testNoiseReplacerExternalFootprints(self, noiseSource, variance):
+        # We choose a random seed which causes the test to pass.
+        task = self.makeSingleFrameMeasurementTask("test_NoiseReplacer")
+        task.config.noiseReplacer.noiseSource = noiseSource
+        exposure, catalog = self.dataset.realize(variance, task.schema, randomSeed=0)
+        if noiseSource == 'meta':
+            md = PropertyList()
+            md['BGMEAN'] = variance
+            exposure.setMetadata(md)
+        footprints = task.getFootprintsFromCatalog(catalog)
+        task.run(catalog, exposure, footprints=footprints)
```

Comment:
Should you also check that the default (`footprints=None`) works?

Reply:
That's already tested in the existing test that this is a copy of except pulling out the footprints.